hatchet/graphframe.py

```diff
@@ -76,6 +76,134 @@ def __init__(self, graph, dataframe, exc_metrics=None, inc_metrics=None):
+                "data": {"type": "array"},
+                "columns": {"type": "array"},
+                "column_metadata": {"type": "array"},
+                "nodes": {"type": "array"},
+            }
+        }
+
+        _SCHEMA_PYINSTRUMENT = {
+            "type": "object",
+             "properties": {
+                "start_time": {"type": "number"},
+                "duration": {"type": "number"},
+                "sample_count": {"type": " number"},
+                "program": {"type": "string"},
+                "cpu_time": {"type": "number"},
+                "root_frame": {"type": "object"}
+            }
+        }
+
+        _SCHEMA_TIMEMORY = {
+            "type": "object",
+             "properties": {
+                "timemory": {"type": "object"}
+            }
+        }
+
+        # Determine dirname_or_data is a str and a path exists
+        if type(dirname_or_data) == 'str' and os.path.exists(os.path.dirname(dirname_or_data)):
+            
+            # check if it is a directory.
+            # Formats in directory: hpctoolkit
+            if os.path.isdir(dirname_or_data):
+                _metric_db_files = [f for f in os.listdir(dirname_or_data) if f.endswith('.metric-db')] 
+                _experiment_xml_files = [f for f in os.listdir(dirname_or_data) if f.endswith('experiment.xml')]
+
+                if len(_metric_db_files) == 0:
+                    raise ValueError("No metric-db files detected inside the provided path.")
+
+                if len(_experiment_xml_files) == 0:
+                    raise ValueError("No experiment.xml file detected inside the provided path.")
+                
+                return 'hpctoolkit'
+                
+
+            # check if it is a file
+            elif os.path.isfile(dirname_or_data):
+                _file_name, _file_ext = os.path.splitext(dirname_or_data)
+
+                if _file_ext == "cali":
+                    return 'caliper'
+                elif _file_ext == "pstats":
+                    return 'pstats'
+                elif _file_ext == 'json':
+                    # TODO: Check if we can just load the key and dtype of JSON.
+                    # We could also be unnecessarily read the data again.
+                    try:
+                        _data=json.loads(dirname_or_data)
+                    except ValueError as err:
+                        print(err)
+                        return False
+
+                    if jsonschema.validate(instance=_data, schema=_SCHEMA_CALIPER_JSON):
+                        return 'caliper_json'
+                    elif jsonschema.validate(instance=_data, schema=_SCHEMA_PYINSTRUMENT):
+                        return 'pyinstrument'
+                    elif jsonschema.validate(instance=_data, schema=_SCHEMA_TIMEMORY):
+                        return 'timemory'
+                elif _file_ext == "dot":
+                    return 'grpof'
+        elif type(dirname_or_data) == 'list':
```

Comment:
i generally like to put simpler case at thee top.. thiis function wont be call many times so there is no performance hit. otherwise, you'd want to put more frequent cases first.